apps/prairielearn/src/pages/news_item/news_item.html.ts

```diff
@@ -0,0 +1,75 @@
+import { z } from 'zod';
+
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { NewsItemSchema } from '../../lib/db-types.js';
+
+export const NewsItemRowSchema = NewsItemSchema.extend({
+  formatted_date: z.string(),
+  show_student_badge: z.boolean(),
```

Comment:
We could do this calculation in JS too: add a single call to `users_is_instructor_in_any_course`, then check `visible_to_students` on each row. Then we can use `NewsItemSchema` directly.

Reply:
Done.